packages/remix-dev/__tests__/replace-remix-imports-test.ts

```diff
@@ -0,0 +1,101 @@
+import path from "path";
+import fse from "fs-extra";
+import os from "os";
+import stripAnsi from "strip-ansi";
+import type { PackageJson } from "@npmcli/package-json";
+
+import { run } from "../cli/run";
+
+let output: string;
+const ORIGINAL_IO = {
+  log: console.log,
+  warn: console.warn,
+  error: console.error,
+};
+function mockLog(message: unknown = "", ...rest: Array<unknown>) {
+  // if you need to debug stuff, then use:
+  // console.log('debug:', 'whatever you need to say');
+  if (typeof message === "string" && message.startsWith("debug:")) {
+    return ORIGINAL_IO.log(message, ...rest);
+  }
+  let messageString =
+    typeof message === "string" ? message : JSON.stringify(message, null, 2);
+  if (rest[0]) {
+    throw new Error(
+      "Our tests are not set up to handle multiple arguments to console.log."
+    );
+  }
+  output += "\n" + stripAnsi(messageString);
+}
+
+const FIXTURE = path.join(__dirname, "fixtures/replace-remix-imports");
+
+const TEMP_DIR = path.join(
+  fse.realpathSync(os.tmpdir()),
+  `remix-tests-${Math.random().toString(32).slice(2)}`
+);
+
+beforeEach(async () => {
+  output = "";
+  console.log = mockLog;
+  console.warn = mockLog;
+  console.error = mockLog;
+
+  await fse.remove(TEMP_DIR);
+  await fse.ensureDir(TEMP_DIR);
+});
+afterEach(async () => {
+  console.log = ORIGINAL_IO.log;
+  console.warn = ORIGINAL_IO.warn;
+  console.error = ORIGINAL_IO.error;
+
+  await fse.remove(TEMP_DIR);
+});
+
+const makeApp = () => {
+  let projectDir = path.join(TEMP_DIR, "replace-remix-imports");
+  fse.copySync(FIXTURE, projectDir);
+  return projectDir;
+};
+
+const migrate = async (projectDir: string) => {
+  await run([
+    "migrate",
+    "--migration",
+    "replace-remix-imports",
+    projectDir,
+    "--force",
+  ]);
+};
+
+describe("`replace-remix-imports` migration", () => {
+  it("runs successfully", async () => {
```

Comment:
Let's add a couple assertions in here that look for the text `from "remix"` and ensures it no longer exists in the codebase.

Reply:
And perhaps we could run the `build` script in the fixture after the migration to make sure imports can be resolved properly.

Reply:
I added checks that `from "remix"` is purged from the app post-migration.

As for running `build`, adding `npm install` and `npm run build` to the test makes the test go from sub-second to more than 1 minute on my machine (M1 Max) and becomes network dependent. I think I prefer leaving the `build` check out and keeping this test lean. What do you think @kentcdodds ?

Reply:
Yeah, definitely don't want it to be network dependent 👍 We could probably simplify the app further (remove all the deps and stuff and make remix reference the local remix), but that's more work than it's worth. This is enough 👍